docs/managing_deis/upgrading-deis.rst

```diff
@@ -63,6 +63,56 @@ Finally, update ``deisctl`` to the new version and reinstall:
     enable PROXY protocol for ports 80 and 443 on the ELB, add a ``TCP 443:443`` listener, and
     change existing targets and health checks from HTTP to TCP.
 
+
+Graceful Upgrade
+----------------
+
+Alternatively, an experimental feature exists to provide the ability to perform a graceful upgrade. Unlike the
+in-place process above, this process keeps the platforms routers and publishers up during the upgrade process.
+This means that there should only be a maximum of around 1-2 seconds of downtime while the routers boot up. Many
```

Comment:
> This means that there should only be a maximum of around 1-2 seconds of downtime while the routers boot up

Is this on account of how health checks from the load balancers are configured?  I can imagine a scenario where even with the rolling restarts, the timing of the health checks that determine a router is down and then back up could have all _n_ routers out of commission for a while.  This would happen if health checks were frequent and quick to consider a router down after a couple bad responses, but slow to consider it back up again by requiring some substantial number of healthy responses first.